assets/app/view/game/part/track_node_path.rb

```diff
@@ -335,16 +351,30 @@ def load_from_tile
             @end_y
           ) if @need_arc
 
-          return if !@path.parallel? || !@begin_edge
+          return if @path.single? || !@begin_edge && !@end_edge
 
-          shift = (@path.lane_index * 2 - @path.lanes + 1) * (@width + PARALLEL_SPACING[@path.lanes - 2]) / 2.0
-          delta_x = (shift * Math.cos(@begin_edge * 60.0 * Math::PI / 180.0)).round(2)
-          delta_y = (shift * Math.sin(@begin_edge * 60.0 * Math::PI / 180.0)).round(2)
+          begin_shift_edge = @begin_edge || @end_edge
+          end_shift_edge = @end_edge || @begin_edge
 
-          @begin_x += delta_x
-          @begin_y += delta_y
-          @end_x += delta_x
-          @end_y += delta_y
+          if lanes.first[0] > 1
```

Comment:
make lanes.first a var or something

first_lane = lanes[0]

Reply:
or since you use both

first, last = lanes

Reply:
OK